src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,996 @@
+                                                            replace_string_int_idx, replace_stringarray_int_idx, &
+                                                            replace_stringlist_int_idx
+    end type stringlist_type
+
+    interface operator(<)
+        module procedure string_lower
+    end interface
+
+    interface operator(>)
+        module procedure string_greater
+    end interface
+
+    interface operator(==)
+        module procedure string_equal
+    end interface
+
+    interface operator(//)
+        module procedure append_string
+        module procedure prepend_string
+        module procedure append_stringlist
+        module procedure append_stringarray
+        module procedure prepend_stringarray
+    end interface
+contains
+
+! stringlist_index_add --
+!     Add an integer offset to the special index
+!
+! Arguments:
+!     index                 Special index
+!     offset                Offset to be added
+!
+function stringlist_index_add( index, offset )
+    type(stringlist_index_type), intent(in) :: index
+    integer, intent(in)                     :: offset
+
+    type(stringlist_index_type)             :: stringlist_index_add
+
+    stringlist_index_add        = index
+    stringlist_index_add%offset = stringlist_index_add%offset + offset
+end function stringlist_index_add
+
+! stringlist_index_subtract --
+!     Subtract an integer offset to the special index
+!
+! Arguments:
+!     index                 Special index
+!     offset                Offset to be subtracted
+!
+function stringlist_index_subtract( index, offset )
+    type(stringlist_index_type), intent(in) :: index
+    integer, intent(in)                     :: offset
+
+    type(stringlist_index_type)             :: stringlist_index_subtract
+
+    stringlist_index_subtract        = index
+    stringlist_index_subtract%offset = stringlist_index_subtract%offset - offset
+end function stringlist_index_subtract
+
+! compare string_type derived types
+!     Required by sorting functions
+!
+elemental logical function string_lower( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_lower = string1%value < string2%value
+end function string_lower
+
+elemental logical function string_greater( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_greater = string1%value > string2%value
+end function string_greater
+
+elemental logical function string_equal( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_equal = string1%value == string2%value
+end function string_equal
+
+function append_string( list, string )
+     type(stringlist_type), intent(in) :: list
+     character(len=*), intent(in)      :: string
+     type(stringlist_type)             :: append_string
+
+     append_string = list
```

Comment:
I have a query in this line:
I think this will create a copy (deep copy) of `list` which will be assigned to `append_string` here. Have I got it right?

Reply:
But then why are you not returning `append_string` as the output of the function? As soon as we exit this function the `append_string` will be lost and `list` will be unaffected.